p2p_wallet/Scenes/Main/History/History.SceneModel.swift

```diff
@@ -0,0 +1,104 @@
+//
+// Created by Giang Long Tran on 12.04.2022.
+//
+
+import BECollectionView
+import FeeRelayerSwift
+import Foundation
+import RxSwift
+import SolanaSwift
+
+extension History {
+    class SceneModel: BEStreamListViewModel<SolanaSDK.ParsedTransaction> {
+        private let disposeBag = DisposeBag()
+
+        private let solanaSDK: SolanaSDK
+        private let walletsRepository: WalletsRepository
+        @Injected private var feeRelayer: FeeRelayerAPIClientType
+
+        /// A list of source, where data can be fetched
+        private var source: StreamSource
+
+        init(
+            solanaSDK: SolanaSDK = Resolver.resolve(),
+            walletsRepository: WalletsRepository = Resolver.resolve()
+        ) {
+            self.solanaSDK = solanaSDK
+            self.walletsRepository = walletsRepository
+
+            let transactionRepository = CachingTransactionRepository(
+                delegate: SolanaTransactionRepository(solanaSDK: solanaSDK)
+            )
+
+            let transactionParser = CachingTransactionParsing(
+                delegate: DefaultTransactionParser(solanaSDK: solanaSDK, p2pFeePayers: Defaults.p2pFeePayerPubkeys)
+            )
+
+            /// Create source
+            source = MultipleAccountsStreamSource(
+                sources: walletsRepository
+                    .getWallets()
+                    .map { wallet in
+                        AccountStreamSource(
+                            account: wallet.pubkey ?? "",
+                            accountSymbol: wallet.token.symbol,
+                            transactionRepository: transactionRepository,
+                            transactionParser: transactionParser
+                        )
+                    }
+            )
+
+            // TODO: Remove - for testing purpose
+            /*
+             if let wallet = walletsRepository.nativeWallet {
+                 source = AccountStreamSource(
+                     account: wallet.pubkey ?? "",
+                     accountSymbol: wallet.token.symbol,
+                     transactionRepository: transactionRepository,
+                     transactionParser: transactionParser
+                 )
+             }
+             */
+
+            super.init(isPaginationEnabled: true, limit: 10)
+        }
+
+        override func clear() {
+            source.reset()
+            super.clear()
+        }
+
+        override func next() -> Observable<[SolanaSDK.ParsedTransaction]> {
+            AsyncThrowingStream<[SolanaSDK.ParsedTransaction], Error> { stream in
```

Comment:
Needs to test if this `SceneModel` is normally deinited

Reply:
As soon user pulls collection to refresh, the current stream will be canceled and be replaced by new one